lib/requester/util.js

```diff
@@ -116,15 +130,23 @@ module.exports = {
      *
      * @todo: Move this to the SDK.
      * @param request
+     * @param options
      */
-    getRequestBody: function (request) {
-        var options,
-            mode = _.get(request, 'body.mode'),
+    getRequestBody: function (request, options) {
+        var mode = _.get(request, 'body.mode'),
             body = _.get(request, 'body'),
-            empty = request.body && request.body.isEmpty && request.body.isEmpty(),
-            content;
+            empty = body ? body.isEmpty() : true,
+            method,
+            content,
+            computedBody;
+
+        if (!request || empty || !mode || !body[mode]) {
```

Comment:
Why do || based checks? It will needlessly evaluate till end. 

Next if body is undefined checking mode in it will cause error.